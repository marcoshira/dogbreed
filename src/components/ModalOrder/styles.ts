/* istanbul ignore file */
import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    img {
      width: ${theme.frameSizes.medium};
      height: ${theme.frameSizes.medium};
    }

    @media ${theme.media.lteOrEqMedium} {
      width: ${theme.frameSizes.largeMedium};
    }

    @media ${theme.media.lteOrEqSmall} {
      width: 100vw;
    }

    @media ${theme.media.lteSmall} {
      width: 100vw;
      height: 55vh;
    }
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    z-index: 6;
    position: fixed;
    top: 2rem;
    right: 2rem;
    width: 3rem;
    height: 3rem;
    background: none;
    color: ${theme.colors.white};
    border: none;
    display: flex;
    align-items: center;
    justify-content: center;
    transition: ${theme.transitions.fastest};
    cursor: pointer;
    > svg {
      width: 3rem;
      height: 3rem;
    }

    &:hover {
      transform: scale(1.05);
      color: ${theme.colors.black};
      /* background: ${theme.colors.white}; */
    }
  `}
`;
