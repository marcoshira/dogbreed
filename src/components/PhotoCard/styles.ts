import styled, { css } from 'styled-components';

export const Wrapper = styled.a`
  ${({ theme }) => css`
    display: inline-block;
    margin: 10px;
    cursor: pointer;
    overflow: hidden;
    height: 260px;
    width: 260px;
    border-radius: 5%;

    img {
      height: 260px;
      width: 260px;
      transition: all 300ms ease-in-out;

      object-fit: cover;

      &:hover {
        transform: scale(1.2) rotate(10deg);
      }
    }
  `}
`;
