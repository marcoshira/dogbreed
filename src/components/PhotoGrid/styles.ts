import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(260px, 1fr));
    gap: 0;
    @media ${theme.media.lteSmall} {
      grid-template-columns: 1fr;
    }
  `}
`;
