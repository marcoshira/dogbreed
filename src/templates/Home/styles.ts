import styled from 'styled-components';
import { Wrapper as Container } from '../../components/Wrapper/styles';

export const Wrapper = styled.div`
  ${Container} {
    max-width: 90rem;
  }
`;
