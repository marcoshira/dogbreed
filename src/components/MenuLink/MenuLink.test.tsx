import { fireEvent, screen, waitFor } from '@testing-library/react';
import { MenuLink } from '.';
import { renderTheme } from '../../styles/render-theme';

const func = jest.fn();

describe('<MenuLink />', () => {
  it('should render a link', () => {
    renderTheme(<MenuLink link="http://localhost">Children</MenuLink>);
    expect(screen.getByText('Children')).toHaveAttribute('target', '_self');
  });

  it('should open in new tab', () => {
    renderTheme(
      <MenuLink link="http://localhost" newTab={true}>
        Children
      </MenuLink>,
    );
    expect(screen.getByText('Children')).toHaveAttribute('target', '_blank');
  });

  it('should call a function', async () => {
    renderTheme(
      <MenuLink link="http://localhost" onClick={func}>
        Children
      </MenuLink>,
    );
    expect(screen.getByText('Children')).toHaveAttribute('target', '_self');
    fireEvent.click(screen.getByText('Children'));
    await waitFor(() => {
      expect(func).toHaveBeenCalled();
    });
  });

  it('should match snapshot', () => {
    renderTheme(<MenuLink link="http://localhost">Children</MenuLink>);
    expect(screen.getByText('Children').parentElement).toMatchSnapshot();
  });
});
