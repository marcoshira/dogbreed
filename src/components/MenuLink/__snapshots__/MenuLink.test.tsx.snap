// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink /> should match snapshot 1`] = `
.c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 2rem;
  padding: 1.6rem;
  color: #ff003c;
  position: relative;
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
  cursor: pointer;
}

.c0:hover {
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
}

.c0 svg {
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
  width: 30px;
  color: #ff003c;
}

.c0 svg:hover {
  -webkit-transform: scale(1.15);
  -ms-transform: scale(1.15);
  transform: scale(1.15);
}

<div>
  <a
    class="c0"
    target="_self"
  >
    Children
  </a>
</div>
`;
